Scripts/RemediationScripts/Remediate-EnableWAFPolicyPreventionModeForFrontDoor.ps1

```diff
@@ -0,0 +1,952 @@
+                                                                            else
+                                                                            {
+                                                                                $false
+                                                                                
+                                                                            }
+                                                                        }
+                                                                        }} 
+            }
+            catch
+            {
+                Write-Host "Error fetching Front Door FrontEndpoint: ID - [$($frontdoorFrontEndpointId)]. Error: $($_)" -ForegroundColor $([Constants]::MessageType.Error)
+                Write-Host "Skipping this Front Door FrontEndpoint..." -ForegroundColor $([Constants]::MessageType.Warning)
+            }
+        }
+    }
+     
+ 
+    if ([String]::IsNullOrWhiteSpace($FilePath))
+    {
+        $totalfrontDoors = ($frontDoors | Measure-Object).Count
+
+        if ($totalfrontDoors -eq 0)
+        {
+            Write-Host "No Front Doors found. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+            Write-Host $([Constants]::DoubleDashLine)
+            break
+        }
+
+        Write-Host "Found [$($totalfrontDoors)] Front Door(s)." -ForegroundColor $([Constants]::MessageType.Update)
+        Write-Host $([Constants]::SingleDashLine)
+    }
+    
+  
+   
+    $totalfrontDoorFrontendPoints = ($frontDoorFrontendPoints | Measure-Object).Count
+
+    if ($totalfrontDoorFrontendPoints -eq 0)
+    {
+        Write-Host "No Front Door Endpoint(s) found with WAF not Configured. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+        Write-Host $([Constants]::DoubleDashLine)
+        break
+    }
+    
+    Write-Host "Found [$($totalfrontDoorFrontendPoints)] Front Door Frontendpoint(s)" -ForegroundColor $([Constants]::MessageType.Update)
+    Write-Host $([Constants]::SingleDashLine)
+
+    # Includes Front Door Endpoint(s) where WAF Policy is not in Prevention Mode
+    $frontDoorEndpointsWithWAFPolicyNotInPrevention = @()
+
+    # Includes Front Door Endpoint(s) that were skipped during remediation. There were errors remediating them.
+    $frontDoorEndpointsSkipped = @()
+
+    
+    
+   
+    Write-Host "[Step 3 of 5] Fetching Frontendpoint(s)..."
+    Write-Host $([Constants]::SingleDashLine)
+    Write-Host "Separating Front Door Endpoint(s) for which WAF Policy is not in Prevention Mode..." -ForegroundColor $([Constants]::MessageType.Info)
+    Write-Host $([Constants]::SingleDashLine)
+
+    $frontDoorFrontendPoints | ForEach-Object {
+        $frontEndPoint = $_        
+            if(($_.IsPreventionMode -eq $false) -and ($_.IsWAFConfigured -eq $true))
+            {
+                $frontDoorEndpointsWithWAFPolicyNotInPrevention += $frontEndPoint
```

Comment:
How we are handling front doors for which itself is not configured.

Reply:
Can we show at least some warning so that end user will know that something is missing and they need to run one another BRS as well?

Reply:
added warning in all BRS where required for configuring WAF Policy